src/components/filters/DateFilter/utils/OptionUtils.ts

```diff
@@ -1,10 +1,23 @@
 // (C) 2007-2019 GoodData Corporation
 import { ExtendedDateFilters } from "@gooddata/typings";
 
+const NUM_OF_MILISECONDS_IN_MINUTE = 60000;
 export const getDateFilterOptionGranularity = (
     dateFilterOption: ExtendedDateFilters.DateFilterOption,
 ): ExtendedDateFilters.DateFilterGranularity =>
     ExtendedDateFilters.isRelativeDateFilterForm(dateFilterOption) ||
     ExtendedDateFilters.isRelativeDateFilterPreset(dateFilterOption)
         ? dateFilterOption.granularity
         : undefined;
+
+export const getTimeOffsetInMilliseconds = (): number =>
+    new Date().getTimezoneOffset() * NUM_OF_MILISECONDS_IN_MINUTE;
+
+// Gets the time value in milliseconds at UTC 0
+// Example: "2019-11-28" with GMT -5(hour) => "2019-11-27T19:00:00Z" + 5(hour) => get time at "2019-11-28 00:00:00 GTM-5"
+export const getTimeValueAtUTC0FromString = (dateString: string): number => {
```

Comment:
This conversion is difficult to understand for me. Maybe I dont understand whole logic but if you need date in UTC why do not use this method.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getUTCDate

Reply:
I have update and add more description hope it's clear, see detail at https://github.com/gooddata/gooddata-react-components/pull/1237/files#diff-31d812ef798c6e87d4e62d7ea5c0adccR18